docs/compare_collections.md

```diff
@@ -56,15 +56,15 @@ Now that you know the basics, and once you have an understanding of what the com
 
 ## Limitation of the comparison function: distinguishing out-of-order from mismatched arrays
 
-One limitation of the comparison function is that it does comparisons at the level of arrays, not at the level of individual elements. What would the comparison function return for two sequence collections that are exact duplicates except for two component sequences that have swapped names?
+One limitation of the comparison function is that it does comparisons at the level of arrays, not at the level of individual elements. What would the comparison function return for two sequence collections that have the same content, but in different orders, AND where in addition two of the sequences have swapped names?
 
 Because the sequence array would contain the same sequences, the comparison function will count them all as matching.
 Similarly, the names arrays contain the same names and so all will be counted as a match.
-However, the same_order will *not* be true, so it will yield false.
+However, the same_order will *not* be true; it will yield false for all attributes.
 
 This is the same output as a comparison of two sequence collections in different orders, without the name swap. This is a fundamental limitation of the array-based method of comparing.
```

Comment:
> This is the same output as a comparison of two sequence collections in different orders, without the name swap.

My main problem is with this line. In the example where everything is the same except for a name swap, then the output will NOT be the same as a comparison of two sequence collections in different order. As you mention above, in the main example, only the `same_order` value for the `names` attribute will be `false`. In the comparison of two sequence collections in different orders, all `same_order` values will be `false`!

Reply:
yes you are right. I think this may be an outdated explanation from before we split the `same_order` into a per-array field.

That removes this particular limitation with the compare function.